// created by Kevin @ 6th Nov. 2017

// provide proto version
syntax = "proto3";

// TODO specify options... e.g.
// option java_package = "JAVA_PACKAGE";


package provenance_system;


// prodide service definition:
service gateway {
  rpc push_data(stream Grid_data) returns (reply) {}
}

// Grid_data message
// composition of meter- and measurement_message
// in compliance with the data model provided from grid data
message Grid_data {
  measurement_message measurement = 1;
}

// measurement_message
message measurement_message {
  int32 meter_id = 1;
  int32 metric_id = 2;
  int32 timestamp = 3;
  string values = 4;
}

message reply {
  string response_code=1;
}
